include/fiction/networks/views/bfs_topo_view.hpp

```diff
@@ -0,0 +1,295 @@
+
+    /**
+     * Constructs a breadth-first topological view from a given network.
+     *
+     * @param ntk Logic network to be wrapped with a topological view.
+     */
+    explicit bfs_topo_view(Ntk const& ntk) : mockturtle::immutable_view<Ntk>(ntk), fanout_ntk(ntk)
+    {
+        static_assert(mockturtle::is_network_type_v<Ntk>, "Ntk is not a network type");
+        static_assert(mockturtle::has_size_v<Ntk>, "Ntk does not implement the size method");
+        static_assert(mockturtle::has_get_constant_v<Ntk>, "Ntk does not implement the get_constant method");
+        static_assert(mockturtle::has_foreach_pi_v<Ntk>, "Ntk does not implement the foreach_pi method");
+        static_assert(mockturtle::has_foreach_fanin_v<Ntk>, "Ntk does not implement the foreach_fanin method");
+        static_assert(mockturtle::has_incr_trav_id_v<Ntk>, "Ntk does not implement the incr_trav_id method");
+        static_assert(mockturtle::has_set_visited_v<Ntk>, "Ntk does not implement the set_visited method");
+        static_assert(mockturtle::has_trav_id_v<Ntk>, "Ntk does not implement the trav_id method");
+        static_assert(mockturtle::has_visited_v<Ntk>, "Ntk does not implement the visited method");
+
+        update_bfs_topo();
+    }
+
+    /**
+     * Returns the number of nodes in the topological view.
+     *
+     * @return Total number of visited nodes in the topological order.
+     */
+    auto size() const
+    {
+        return static_cast<uint32_t>(topo_order.size());
+    }
+
+    /**
+     * Returns the number of gates in the topological view.
+     *
+     * @return Number of gates excluding constants and primary inputs.
+     */
+    auto num_gates() const
+    {
+        uint32_t const offset = 1u + this->num_pis() +
+                                (this->get_node(this->get_constant(true)) != this->get_node(this->get_constant(false)));
+        return static_cast<uint32_t>(topo_order.size() - offset);
+    }
+
+    /**
+     * Returns the topological index of a node.
+     *
+     * @param n Node whose index is to be determined.
+     * @return Index of the node in the topological order.
+     */
+    uint32_t node_to_index(node const& n) const
+    {
+        return std::distance(std::begin(topo_order), std::find(std::begin(topo_order), std::end(topo_order), n));
+    }
+
+    /**
+     * Returns the node at a given topological index.
+     *
+     * @param index Index into the topological order.
+     * @return Node at the specified index.
+     */
+    node index_to_node(uint32_t index) const
+    {
+        return topo_order.at(index);
+    }
+
+    /**
+     * Iterates over all nodes in topological order.
+     *
+     * @tparam Fn Callable function or lambda to be applied to each node.
+     * @param fn Function to apply to each node.
+     */
+    template <typename Fn>
+    void foreach_node(Fn&& fn) const
+    {
+        mockturtle::detail::foreach_element(topo_order.begin(), topo_order.end(), fn);
```

Comment:
The ranges can be `const`:
```suggestion
        mockturtle::detail::foreach_element(topo_order.cbegin(), topo_order.cend(), fn);
```

Reply:
Analogously for the other functions.